### DISCUSSION TITLE
Removing the Telescope extension in favor of `vim.ui.select`.

### DISCUSSION THREAD

**Discussion by ckipp01** [scalameta/nvim-metals#480] [Polls]
Hey folks, as you probably know we have a Telescope extension in https://github.com/scalameta/nvim-metals/tree/main/lua/telescope/_extensions which I personally make heavy use of to quickly filter through and select a Metals command. However I added this early on before `vim.ui.select` existed in Neovim.  If you're unfamiliar with `vim.ui.select` check out `:h vim.ui` but it's a generalized way to offer selections and input that other plugins can implement. So basically if you use something like https://github.com/stevearc/dressing.nvim you could get our extension for free if I was using `vim.ui.select`. I'm thinking of refactoring the commands to use `vim.ui.select` to offer more flexibility. It would also mean we wouldn't have to maintain the Telescope extension.

**Comment by lenguyenthanh** [comment]
There is also [`telescope` plugin](https://github.com/nvim-telescope/telescope-ui-select.nvim) for `vim.ui` so I think this is a good idea to support more general way and remove external dependency.

**Comment by ckipp01** [comment]
So I was just able to play around with this. Right now the telescope extension looks like this:

<img width="658" alt="Screenshot 2022-10-31 at 16 05 27" src="https://user-images.githubusercontent.com/13974112/199040786-f0ca59be-6909-4113-a0a2-e93782ddc19b.png">

But when I switch it over to use `vim.ui.select` with dressing.nvim it looks like this:

<img width="658" alt="Screenshot 2022-10-31 at 16 05 34" src="https://user-images.githubusercontent.com/13974112/199040875-4f7a046e-772e-4f32-9633-191064fa22dd.png">

So it's for sure now as nice, but more flexible. I'm unsure what to do because I like the way this is done. I might just add the new way and then leave the telescope in there as well so that if a user wants to use it they can or they can use the `vim.ui.select` way.

**Reply by ckipp01** [reply]
https://github.com/scalameta/nvim-metals/pull/486


![image](https://user-images.githubusercontent.com/13974112/199043917-3903d3b3-f30c-40d8-ba61-a0aa1fd7a1ea.png)
